riotfile.py

```diff
@@ -1,9 +1,52 @@
 # type: ignore
+import json
+import logging
+import os
+import sys
 from typing import List
 from typing import Tuple
+from urllib.error import HTTPError
+from urllib.request import urlopen
 
 from riot import Venv
-from riot import latest
+from riot import latest as latest_riot
+
+
+logger = logging.getLogger(__name__)
+latest = object()  # sentinel value
+
+
+def find_workflow(workflow_id):
+    """Use CircleCI API to retrieve current workflow name"""
+    try:
+        url = f"https://circleci.com/api/v2/workflow/{workflow_id}"
+        res = urlopen(url)
+        body = res.read().decode()
+        return json.loads(body)["name"]
+    except HTTPError:
+        logger.error("Error loading workflow information from the CircleC: %s", url)
+        raise
+
+
+# Set up PY_Latest to True if the current workflow is test_latest
+PY_Latest = False
```

Comment:
It would require to change all calls to riot and add one more script. I tried to make this PR the less invasive possible and changing all riot calls by pre-riot seemed more problematic. But it's possible.

Reply:
Please tell me what you decide on this topic.

Reply:
I think that it's reasonable to run a pre script to gather information about the environment to run riot. One more script here isn't a bad thing imo.